Docker Build - Failing

```
>docker build echidna
[+] Building 2.0s (12/21)
 => [internal] load build definition from Dockerfile                                                                                                                                                          0.1s
 => => transferring dockerfile: 1.64kB                                                                                                                                                                        0.0s
 => [internal] load .dockerignore                                                                                                                                                                             0.1s
 => => transferring context: 34B                                                                                                                                                                              0.0s
 => [internal] load metadata for docker.io/library/ubuntu:bionic                                                                                                                                              1.4s
 => [internal] load build context                                                                                                                                                                             0.1s
 => => transferring context: 14.21kB                                                                                                                                                                          0.0s
 => [builder 1/8] FROM docker.io/library/ubuntu:bionic@sha256:9bc830af2bef73276515a29aa896eedfa7bdf4bdbc5c1063b4c457a4bbb8cd79                                                                                0.0s
 => CACHED [final  2/10] WORKDIR /root                                                                                                                                                                        0.0s
 => CACHED [builder 2/8] RUN apt-get update && apt-get -y upgrade && apt-get install -y sudo git cmake curl libgmp-dev libssl-dev libbz2-dev libreadline-dev software-properties-common libsecp256k1-dev &&   0.0s
 => CACHED [builder 3/8] RUN sudo apt-get install -y build-essential                                                                                                                                          0.0s
 => CACHED [builder 4/8] RUN wget -qO- https://get.haskellstack.org/ | sh                                                                                                                                     0.0s
 => CACHED [builder 5/8] COPY . /echidna/                                                                                                                                                                     0.0s
 => CACHED [builder 6/8] WORKDIR /echidna                                                                                                                                                                     0.0s
 => ERROR [builder 7/8] RUN .github/scripts/install-libff.sh                                                                                                                                                  0.4s
------
 > [builder 7/8] RUN .github/scripts/install-libff.sh:
#12 0.319 /bin/sh: 1: .github/scripts/install-libff.sh: not found
------
executor failed running [/bin/sh -c .github/scripts/install-libff.sh]: exit code: 127
```

Hi @shreyas3336, I just tried building the repo `Dockerfile` on the `master` branch and I was not able to reproduce the issue. Can you provide more details on the issue you are experiencing?

```shell
~$ git clone https://github.com/crytic/echidna
Cloning into 'echidna'...
remote: Enumerating objects: 8619, done.
remote: Counting objects: 100% (953/953), done.
remote: Compressing objects: 100% (494/494), done.
remote: Total 8619 (delta 525), reused 821 (delta 434), pack-reused 7666
Receiving objects: 100% (8619/8619), 1.65 MiB | 12.67 MiB/s, done.
Resolving deltas: 100% (5132/5132), done.
~$ cd echidna/
~/echidna$ ls
CHANGELOG.md  CONTRIBUTING.md  Dockerfile  LICENSE  README.md  action  default.nix  echidna.png  examples  lib  package.yaml  src  stack.yaml
~/echidna$ docker build .
Sending build context to Docker daemon  951.8kB
Step 1/21 : FROM ubuntu:bionic AS builder
bionic: Pulling from library/ubuntu
284055322776: Already exists 
Digest: sha256:0fedbd5bd9fb72089c7bbca476949e10593cebed9b1fb9edf5b79dbbacddd7d6
Status: Downloaded newer image for ubuntu:bionic
 ---> 5a214d77f5d7
Step 2/21 : ENV LD_LIBRARY_PATH=/usr/local/lib PREFIX=/usr/local HOST_OS=Linux
 ---> Running in 419aa34bb8af
Removing intermediate container 419aa34bb8af
 ---> 27157f58df8e
Step 3/21 : RUN apt-get update && apt-get -y upgrade && apt-get install -y sudo git cmake curl libgmp-dev libssl-dev libbz2-dev libreadline-dev software-properties-common libsecp256k1-dev && apt-get clean
 ---> Running in 5adcbd1e51ef
Get:1 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic InRelease [242 kB]
Get:3 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [2461 kB]
Get:4 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]
Get:5 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [26.8 kB]
Get:6 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [691 kB]
Get:7 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1446 kB]
Get:8 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]
Get:9 http://archive.ubuntu.com/ubuntu bionic/universe amd64 Packages [11.3 MB]
Get:10 http://archive.ubuntu.com/ubuntu bionic/multiverse amd64 Packages [186 kB]
Get:11 http://archive.ubuntu.com/ubuntu bionic/main amd64 Packages [1344 kB]
Get:12 http://archive.ubuntu.com/ubuntu bionic/restricted amd64 Packages [13.5 kB]
Get:13 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [2898 kB]
Get:14 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [725 kB]
Get:15 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [34.4 kB]
Get:16 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2227 kB]
Get:17 http://archive.ubuntu.com/ubuntu bionic-backports/universe amd64 Packages [11.4 kB]
Get:18 http://archive.ubuntu.com/ubuntu bionic-backports/main amd64 Packages [11.3 kB]
Fetched 23.9 MB in 4s (5646 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
Calculating upgrade...
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  binutils binutils-common binutils-x86-64-linux-gnu bzip2-doc ca-certificates
(...)
Setting up cmake (3.10.2-1ubuntu2.18.04.2) ...
Setting up git (1:2.17.1-1ubuntu0.9) ...
Setting up curl (7.58.0-2ubuntu3.16) ...
Setting up gnupg (2.2.4-1ubuntu1.4) ...
Processing triggers for libc-bin (2.27-3ubuntu1.4) ...
Processing triggers for ca-certificates (20210119~18.04.2) ...
Updating certificates in /etc/ssl/certs...
0 added, 0 removed; done.
Running hooks in /etc/ca-certificates/update.d...
done.
Processing triggers for dbus (1.12.2-1ubuntu1.2) ...
Removing intermediate container 5adcbd1e51ef
 ---> dd2a1cbdd2ef
Step 4/21 : RUN curl -sSL https://get.haskellstack.org/ | sh
 ---> Running in 5495522c2128
Detected Linux distribution: ubuntu

Installing dependencies...


About to use 'sudo' to run the following command as root:
    apt-get install -y g++ libffi-dev zlib1g-dev
in order to install required system dependencies.

Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  g++-7 libstdc++-7-dev
Suggested packages:
  g++-multilib g++-7-multilib gcc-7-doc libstdc++6-7-dbg libstdc++-7-doc
The following NEW packages will be installed:
  g++ g++-7 libffi-dev libstdc++-7-dev zlib1g-dev
0 upgraded, 5 newly installed, 0 to remove and 0 not upgraded.
Need to get 11.5 MB of archives.
After this operation, 43.2 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 libstdc++-7-dev amd64 7.5.0-3ubuntu1~18.04 [1471 kB]
Get:2 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 g++-7 amd64 7.5.0-3ubuntu1~18.04 [9697 kB]
Get:3 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 g++ amd64 4:7.4.0-1ubuntu2.3 [1568 B]
Get:4 http://archive.ubuntu.com/ubuntu bionic/main amd64 zlib1g-dev amd64 1:1.2.11.dfsg-0ubuntu2 [176 kB]
Get:5 http://archive.ubuntu.com/ubuntu bionic/main amd64 libffi-dev amd64 3.2.1-8 [156 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 11.5 MB in 2s (5816 kB/s)
Selecting previously unselected package libstdc++-7-dev:amd64.
(Reading database ... 17784 files and directories currently installed.)
Preparing to unpack .../libstdc++-7-dev_7.5.0-3ubuntu1~18.04_amd64.deb ...
Unpacking libstdc++-7-dev:amd64 (7.5.0-3ubuntu1~18.04) ...
Selecting previously unselected package g++-7.
Preparing to unpack .../g++-7_7.5.0-3ubuntu1~18.04_amd64.deb ...
Unpacking g++-7 (7.5.0-3ubuntu1~18.04) ...
Selecting previously unselected package g++.
Preparing to unpack .../g++_4%3a7.4.0-1ubuntu2.3_amd64.deb ...
Unpacking g++ (4:7.4.0-1ubuntu2.3) ...
Selecting previously unselected package zlib1g-dev:amd64.
Preparing to unpack .../zlib1g-dev_1%3a1.2.11.dfsg-0ubuntu2_amd64.deb ...
Unpacking zlib1g-dev:amd64 (1:1.2.11.dfsg-0ubuntu2) ...
Selecting previously unselected package libffi-dev:amd64.
Preparing to unpack .../libffi-dev_3.2.1-8_amd64.deb ...
Unpacking libffi-dev:amd64 (3.2.1-8) ...
Setting up libffi-dev:amd64 (3.2.1-8) ...
Setting up libstdc++-7-dev:amd64 (7.5.0-3ubuntu1~18.04) ...
Setting up zlib1g-dev:amd64 (1:1.2.11.dfsg-0ubuntu2) ...
Setting up g++-7 (7.5.0-3ubuntu1~18.04) ...
Setting up g++ (4:7.4.0-1ubuntu2.3) ...
update-alternatives: using /usr/bin/g++ to provide /usr/bin/c++ (c++) in auto mode
update-alternatives: warning: skip creation of /usr/share/man/man1/c++.1.gz because associated file /usr/share/man/man1/g++.1.gz (of link group c++) doesn't exist

Using generic bindist...

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   138  100   138    0     0    597      0 --:--:-- --:--:-- --:--:--   597
100   637  100   637    0     0   2068      0 --:--:-- --:--:-- --:--:--  2068
100 17.1M  100 17.1M    0     0  26.9M      0 --:--:-- --:--:-- --:--:-- 26.9M
Installing Stack to: /usr/local/bin/stack...

-------------------------------------------------------------------------------

Stack has been installed to: /usr/local/bin/stack

WARNING: '/root/.local/bin' is not on your PATH.
    Stack will place the binaries it builds in '/root/.local/bin' so
    for best results, please add it to the beginning of PATH in your profile.

Removing intermediate container 5495522c2128
 ---> a1906bd628ee
Step 5/21 : COPY . /echidna/
 ---> f353b82fef96
Step 6/21 : WORKDIR /echidna
 ---> Running in 259a54bdfc0a
Removing intermediate container 259a54bdfc0a
 ---> 65dc51cbbb4d
Step 7/21 : RUN .github/scripts/install-libff.sh
 ---> Running in c08947e69b24
+ source .github/scripts/host.sh
++ '[' -z x ']'
++ '[' Linux = Linux ']'
++ EXT=.a
+ '[' -f /root/.local/lib/libff.a ']'
+ git clone https://github.com/scipr-lab/libff --recursive
Cloning into 'libff'...
Submodule 'depends/ate-pairing' (https://github.com/herumi/ate-pairing.git) registered for path 'depends/ate-pairing'
Submodule 'gtest' (https://github.com/google/googletest.git) registered for path 'depends/gtest'
Submodule 'depends/xbyak' (https://github.com/herumi/xbyak.git) registered for path 'depends/xbyak'
Cloning into '/echidna/libff/depends/ate-pairing'...
Cloning into '/echidna/libff/depends/gtest'...
Cloning into '/echidna/libff/depends/xbyak'...
Submodule path 'depends/ate-pairing': checked out 'e69890125746cdaf25b5b51227d96678f76479fe'
Submodule path 'depends/gtest': checked out 'a325ad2db5deb623eab740527e559b81c0f39d65'
Submodule path 'depends/xbyak': checked out 'f0a8f7faa27121f28186c2a7f4222a9fc66c283d'
+ cd libff
+ git submodule init
+ git submodule update
+ git checkout v0.2.1
warning: unable to rmdir 'depends/gtest': Directory not empty
Note: checking out 'v0.2.1'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 5835b8c Merge branch 'clearmatics-fix-curves-interfaces-inconsistencies'
+ ARGS='-DCMAKE_INSTALL_PREFIX=/usr/local -DWITH_PROCPS=OFF'
+ CXXFLAGS=
+ '[' Linux = macOS ']'
+ mkdir build
+ cd build
+ CXXFLAGS='-fPIC '
+ cmake -DCMAKE_INSTALL_PREFIX=/usr/local -DWITH_PROCPS=OFF ..
-- The C compiler identification is GNU 7.5.0
-- The CXX compiler identification is GNU 7.5.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- GMP: /usr/lib/x86_64-linux-gnu/libgmp.so, /usr/include/x86_64-linux-gnu
-- Found OpenSSL: /usr/lib/x86_64-linux-gnu/libcrypto.so (found version "1.1.1") 
-- Configuring done
-- Generating done
-- Build files have been written to: /echidna/libff/build
+ make
Scanning dependencies of target zm
[  2%] Building CXX object depends/CMakeFiles/zm.dir/ate-pairing/src/zm.cpp.o
[  5%] Building CXX object depends/CMakeFiles/zm.dir/ate-pairing/src/zm2.cpp.o
[  8%] Linking CXX static library libzm.a
[  8%] Built target zm
Scanning dependencies of target ff
[ 11%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_g1.cpp.o
[ 14%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_g2.cpp.o
[ 17%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_init.cpp.o
[ 20%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_pairing.cpp.o
[ 23%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/alt_bn128/alt_bn128_pp.cpp.o
[ 26%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_g1.cpp.o
[ 29%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_g2.cpp.o
[ 32%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_init.cpp.o
[ 35%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_pairing.cpp.o
[ 38%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/edwards/edwards_pp.cpp.o
[ 41%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_g1.cpp.o
[ 44%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_g2.cpp.o
[ 47%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_init.cpp.o
[ 50%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_pairing.cpp.o
[ 52%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt4/mnt4_pp.cpp.o
[ 55%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt46_common.cpp.o
[ 58%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_g1.cpp.o
[ 61%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_g2.cpp.o
[ 64%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_init.cpp.o
[ 67%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_pairing.cpp.o
[ 70%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/mnt/mnt6/mnt6_pp.cpp.o
[ 73%] Building CXX object libff/CMakeFiles/ff.dir/common/double.cpp.o
[ 76%] Building CXX object libff/CMakeFiles/ff.dir/common/profiling.cpp.o
/echidna/libff/libff/common/profiling.cpp: In function 'void libff::print_mem(const string&)':
/echidna/libff/libff/common/profiling.cpp:333:35: warning: unused parameter 's' [-Wunused-parameter]
 void print_mem(const std::string &s)
                                   ^
[ 79%] Building CXX object libff/CMakeFiles/ff.dir/common/utils.cpp.o
[ 82%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_g1.cpp.o
[ 85%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_g2.cpp.o
[ 88%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_gt.cpp.o
[ 91%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_init.cpp.o
[ 94%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_pairing.cpp.o
[ 97%] Building CXX object libff/CMakeFiles/ff.dir/algebra/curves/bn128/bn128_pp.cpp.o
[100%] Linking CXX static library libff.a
[100%] Built target ff
+ make install
[  8%] Built target zm
[100%] Built target ff
Install the project...
-- Install configuration: "RelWithDebInfo"
-- Installing: /usr/local/include/libff
-- Installing: /usr/local/include/libff/algebra
-- Installing: /usr/local/include/libff/algebra/fields
-- Installing: /usr/local/include/libff/algebra/fields/fp6_2over3.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp_aux.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp3.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp6_2over3.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp4.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp3.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp12_2over3over2.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp.tcc
-- Installing: /usr/local/include/libff/algebra/fields/field_utils.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp2.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp12_2over3over2.tcc
-- Installing: /usr/local/include/libff/algebra/fields/bigint.tcc
-- Installing: /usr/local/include/libff/algebra/fields/field_utils.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp4.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp2.tcc
-- Installing: /usr/local/include/libff/algebra/fields/fp6_3over2.hpp
-- Installing: /usr/local/include/libff/algebra/fields/fp.hpp
(...)
```

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,1 +1,2 @@
 *.sol linguist-language=Solidity
+*.sh            eol=lf

